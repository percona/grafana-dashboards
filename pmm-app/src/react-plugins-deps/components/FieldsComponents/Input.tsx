import React from 'react';
// import { useField } from 'react-final-form-hooks';
interface InputFieldInterface {
  required?: boolean;
  name?: string;
  'data-cy'?: string;
  label?: string;
  prefix?: string;
  placeholder?: string;
  form?: any;
  style?: any;
  text?: string;
  options?: any;
  defaultValue?: any;
}

export const InputField = ({ name, placeholder, required, form }: InputFieldInterface) => {
  // // @ts-ignore
  // const field = useField(name, form);
  // {...field.input}
  return <input type="text" placeholder={placeholder} className="input-field input-field--dark" />;
};
