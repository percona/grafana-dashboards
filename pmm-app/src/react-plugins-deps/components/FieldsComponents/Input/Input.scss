.input-field-wrapper {
  display: flex;
  flex-direction: column;
  align-content: flex-start;
}

.input-field-wrapper .error-message {
  color: indianred;
}
