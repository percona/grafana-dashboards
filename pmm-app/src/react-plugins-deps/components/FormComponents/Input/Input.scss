#antd {
  .input-field-wrapper {
    display: flex;
    flex-direction: column;
    align-content: flex-start;
    width: 100%;
  }

  .input-field-wrapper .error-message {
    color: indianred;
  }

  .input-field-wrapper .input-field {
    height: 40px;
  }
}
