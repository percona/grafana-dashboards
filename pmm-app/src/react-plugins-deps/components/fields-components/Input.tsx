import React, { FunctionComponent } from 'react';
import { useForm, useField } from 'react-final-form-hooks';

export const InputField = ({ name, placeholder, required, form, ...rest }) => {
  // @ts-ignore
  const field = useField(name, form);

  return <input type="text" {...field.input} placeholder={placeholder} className="input-field input-field--dark" />;
};
