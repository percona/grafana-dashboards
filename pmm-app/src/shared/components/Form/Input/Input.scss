#antd {
  .input-field-wrapper {
    display: flex;
    flex-direction: column;
    align-content: flex-start;
    width: 100%;
  }

  .input-field-wrapper .error-message {
    color: indianred;
  }

  .input-field-wrapper .input-field {
    height: 40px;
  }

  .input-field {
    width: 100%;
    padding: 0 var(--input-lr-inner-indent);
    color: inherit;
    border: 1px solid var(--main-border-color);
    background-color: transparent;
  }

  .input-field::placeholder {
    font-style: italic;
    color: rgb(145, 145, 145);
  }

  .input-field:active,
  .input-field:focus {
    outline: none;
  }

  .input-field--dark {
    background-color: var(--page-background);
  }

  .input--invalid {
    border-color: rgb(255, 0, 0);
  }

  .input-field--interval,
  .input-field--select {
    height: 40px;
    padding: var(--input-lr-inner-indent);
  }

  .input-field--interval {
    width: 80px;
  }

  .input-field--select {
    width: 200px;
  }
}
