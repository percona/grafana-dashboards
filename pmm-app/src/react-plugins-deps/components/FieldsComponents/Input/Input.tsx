import React from 'react';
import { useField } from 'react-final-form-hooks';
import './Input.scss';
interface InputFieldInterface {
  required?: boolean;
  name: string;
  'data-cy'?: string;
  label?: string;
  prefix?: string;
  placeholder?: string;
  form: any;
  style?: any;
  wrapperStyle?: any;
  text?: string;
  options?: any;
  defaultValue?: any;
}

export const InputField = ({ name, placeholder, required, form, style, wrapperStyle }: InputFieldInterface) => {
  // // @ts-ignore
  const { input, meta } = useField(name, form);
  return (
    <div className={'input-field-wrapper'} style={wrapperStyle || {}}>
      <input {...input} type="text" required={required} placeholder={placeholder} style={style || {}} className="input-field input-field--dark" />
      {meta.error && meta.touched && <span className={'error-message'}>{meta.error}</span>}
    </div>
  );
};
