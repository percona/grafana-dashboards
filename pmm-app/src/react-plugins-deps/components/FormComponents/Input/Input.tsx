import React from 'react';
import { useField } from 'react-final-form';
import './Input.scss';
import cx from 'classnames';
interface InputFieldInterface {
  required?: boolean;
  name: string;
  'data-cy'?: string;
  label?: string;
  prefix?: string;
  placeholder?: string;
  style?: any;
  wrapperStyle?: any;
  text?: string;
  options?: any;
  defaultValue?: any;
  readonly?: boolean;
  validate?: any;
  className?: any;
}

export const InputField = ({
  name,
  placeholder = '',
  required,
  style,
  wrapperStyle,
  readonly,
  validate,
  className,
}: InputFieldInterface) => {
  // // @ts-ignore
  const { input, meta } = useField(name, { validate });
  return (
    <div className="input-field-wrapper" style={wrapperStyle || {}}>
      <input
        {...input}
        type="text"
        required={required}
        placeholder={`${required ? '*' : ''}${placeholder}`}
        style={style || {}}
        readOnly={readonly}
        className={cx('input-field input-field--dark', className)}
      />
      {meta.error && meta.touched && <span className="error-message">{meta.error}</span>}
    </div>
  );
};
