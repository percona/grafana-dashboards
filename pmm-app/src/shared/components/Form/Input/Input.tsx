import React from 'react';
import { useField } from 'react-final-form';
import { cx } from 'emotion';
import './Input.scss';

interface InputFieldInterface {
  required?: boolean;
  name: string;
  dataQa?: string;
  label?: string;
  prefix?: string;
  placeholder?: string;
  style?: any;
  wrapperStyle?: any;
  text?: string;
  options?: any;
  defaultValue?: any;
  readonly?: boolean;
  validate?: any;
  className?: any;
}

export const InputField = ({
  name,
  placeholder = '',
  required,
  style,
  wrapperStyle,
  readonly,
  validate,
  className,
  dataQa
}: InputFieldInterface) => {
  const { input, meta } = useField(name, { validate });

  return (
    <div className="input-field-wrapper" style={wrapperStyle || {}}>
      <input
        {...input}
        type="text"
        required={required}
        placeholder={`${required ? '*' : ''}${placeholder}`}
        style={style || {}}
        data-qa={dataQa}
        readOnly={readonly}
        className={cx('input-field input-field--dark', className)}
      />
      {meta.error && meta.touched && <span className="error-message">{meta.error}</span>}
    </div>
  );
};
