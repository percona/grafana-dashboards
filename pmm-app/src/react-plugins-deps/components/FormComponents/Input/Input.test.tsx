import React from 'react';
import { mount } from 'enzyme';
import { Form } from 'react-final-form';
import { InputField } from './Input';

describe('Input field::', () => {
  it('Input renders without props', () => {
    const root = mount(<Form onSubmit={jest.fn()} render={() => <InputField name="testField" />} />);

    expect(root.find('input[name="testField"]')).toHaveLength(1);
    expect(root).toMatchSnapshot();

    root.unmount();
  });
});
