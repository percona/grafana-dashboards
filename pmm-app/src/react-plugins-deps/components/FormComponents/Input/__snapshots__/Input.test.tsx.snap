// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input field:: Input renders without props 1`] = `
<ReactFinalForm
  onSubmit={[MockFunction]}
  render={[Function]}
>
  <Component
    name="testField"
  >
    <div
      className="input-field-wrapper"
      style={Object {}}
    >
      <input
        className="input-field input-field--dark"
        name="testField"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=""
        style={Object {}}
        type="text"
        value=""
      />
    </div>
  </Component>
</ReactFinalForm>
`;
